arch/risc-v/src/common/espressif/Make.defs

```diff
@@ -146,7 +146,10 @@ endif
 
 chip/$(ESP_HAL_3RDPARTY_REPO):
 	$(Q) echo "Cloning Espressif HAL for 3rd Party Platforms"
-	$(Q) git clone --quiet $(ESP_HAL_3RDPARTY_URL) chip/$(ESP_HAL_3RDPARTY_REPO)
+	$(Q) if [ ! -d "$(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO)" ]; then \
+		git clone --quiet $(ESP_HAL_3RDPARTY_URL) $(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO); \
+	 fi
+	$(Q) cp -fr $(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO) chip/$(ESP_HAL_3RDPARTY_REPO)
```

Comment:
Let us combine the logic with the one in https://github.com/apache/nuttx/pull/13297. See: if we don't run `./tools/configure.sh -S` between two different NuttX versions, the build system will try to use the existing folder at the temp folder, but a different version may be required and this can be very dangerous (and hard to debug). So, we should an add an `else` to this clause to check if the version is the same.

Reply:
Or any equivalent logic: if the package exists and the version is the same, use it. Otherwise git clone it again.